cmake/extensions.cmake

```diff
@@ -1002,3 +1003,34 @@ macro(assert_with_usage test comment)
   endif()
 endmacro()
 
+# 3.5. File system management
+function(check_if_directory_is_writeable dir ok)
+  execute_process(
+    COMMAND
+    ${CMAKE_COMMAND} -E touch ${dir}/.zephyr_permission_test.txt
```

Comment:
Minor point, but something like the following one-liner could also work without creating a file, which is friendlier in case of a Control-C:

```
python3 -c 'import os, sys; sys.exit(0 if os.access("/path/to/directory", os.W_OK) else 1)'
```

Reply:
Much better!

https://github.com/zephyrproject-rtos/zephyr/pull/7102/commits/8f8cf6ed7a78ba8c0555d620f096ab32f3c2a9dc

Will squash later before the merge.

Decided to put the python in a script in case someone needs to debug or extend it later.